engine/src/main/java/org/terasology/math/ChunkMath.java

```diff
@@ -0,0 +1,251 @@
+    }
+
+    public static Vector3i calcChunkPos(Vector3i pos) {
+        return calcChunkPos(pos.x, pos.y, pos.z);
+    }
+
+    public static Vector3i calcChunkPos(int x, int y, int z) {
+        return calcChunkPos(x, y, z, ChunkConstants.CHUNK_POWER);
+    }
+
+    public static Vector3i[] calcChunkPos(Region3i region) {
+        return calcChunkPos(region, ChunkConstants.CHUNK_POWER);
+    }
+
+    public static Vector3i calcChunkPos(int x, int y, int z, Vector3i chunkPower) {
+        return new Vector3i(calcChunkPosX(x, chunkPower.x), calcChunkPosY(y, chunkPower.y), calcChunkPosZ(z, chunkPower.z));
+    }
+
+    public static Vector3i[] calcChunkPos(Region3i region, Vector3i chunkPower) {
+        int minX = calcChunkPosX(region.minX(), chunkPower.x);
+        int minY = calcChunkPosY(region.minY(), chunkPower.y);
+        int minZ = calcChunkPosZ(region.minZ(), chunkPower.z);
+
+        int maxX = calcChunkPosX(region.maxX(), chunkPower.x);
+        int maxY = calcChunkPosY(region.maxY(), chunkPower.y);
+        int maxZ = calcChunkPosZ(region.maxZ(), chunkPower.z);
+
+        int size = (maxX - minX + 1) * (maxY - minY + 1) * (maxZ - minZ + 1);
+
+        Vector3i[] result = new Vector3i[size];
+        int index = 0;
+        for (int x = minX; x <= maxX; x++) {
+            for (int y = minY; y <= maxY; y++) {
+                for (int z = minZ; z <= maxZ; z++) {
+                    result[index++] = new Vector3i(x, y, z);
+                }
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Returns the internal position of a block within a chunk.
+     *
+     * @param blockX The X-coordinate of the block in the world
+     * @return The X-coordinate of the block within the chunk
+     */
+    public static int calcBlockPosX(int blockX, int chunkPosFilterX) {
+        return blockX & chunkPosFilterX;
+    }
+
+    public static int calcBlockPosY(int blockY, int chunkPosFilterY) {
+        return blockY & chunkPosFilterY;
+    }
+
+    /**
+     * Returns the internal position of a block within a chunk.
+     *
+     * @param blockZ The Z-coordinate of the block in the world
+     * @return The Z-coordinate of the block within the chunk
+     */
+    public static int calcBlockPosZ(int blockZ, int chunkPosFilterZ) {
+        return blockZ & chunkPosFilterZ;
+    }
+
+    public static Vector3i calcBlockPos(Vector3i worldPos) {
+        return calcBlockPos(worldPos.x, worldPos.y, worldPos.z, ChunkConstants.INNER_CHUNK_POS_FILTER);
+    }
+
+    public static Vector3i calcBlockPos(int x, int y, int z) {
+        return calcBlockPos(x, y, z, ChunkConstants.INNER_CHUNK_POS_FILTER);
+    }
+
+    public static Vector3i calcBlockPos(int x, int y, int z, Vector3i chunkFilterSize) {
+        return new Vector3i(calcBlockPosX(x, chunkFilterSize.x), calcBlockPosY(y, chunkFilterSize.y), calcBlockPosZ(z, chunkFilterSize.z));
+    }
+
+    public static Region3i getChunkRegionAroundWorldPos(Vector3i pos, int extent) {
```

Comment:
I think this could be more immediately clearer if the input variables were posInBlocks and extentInChunks, because I tend to think in blocks, but it seems to me that the method really deals mostly with chunks.